lib/private/Security/Normalizer/IpAddress.php

```diff
@@ -92,6 +92,39 @@ private function getIPv6Subnet(string $ip, int $maskBits = 48): string {
 		return \inet_ntop($binary).'/'.$maskBits;
 	}
 
+	/**
+	 * Returns the IPv4 address embedded in an IPv6 if applicable.
+	 * The detected format is "::ffff:x.x.x.x" using the binary form.
+	 *
+	 * @param string $ipv6 IPv6 string
+	 * @return embedded IPv4 string or false if none was found
```

Comment:
```suggestion
	 * @return string|false embedded IPv4 string or false if none was found
```

Or a bit cleaner maybe a nullable string instead of the mixed type

Reply:
Seems Côme was faster